app/src/main/java/com/gholem/moneylab/features/chart/viewmodel/ChartViewModel.kt

```diff
@@ -0,0 +1,96 @@
+package com.gholem.moneylab.features.chart.viewmodel
+
+import android.os.Build
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.gholem.moneylab.arch.nav.NavigationLiveData
+import com.gholem.moneylab.domain.model.TransactionCategoryModel
+import com.gholem.moneylab.domain.model.TransactionModel
+import com.gholem.moneylab.features.add.domain.GetTransactionListUseCase
+import com.gholem.moneylab.features.add.domain.InsertTransactionModelUseCase
+import com.gholem.moneylab.features.chart.adapter.item.ChartCategoryModel
+import com.gholem.moneylab.features.chart.adapter.item.ChartItem
+import com.gholem.moneylab.features.chart.domain.FetchTransactionModelUseCase
+import com.gholem.moneylab.features.chart.navigation.ChartNavigationEvent
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class ChartViewModel @Inject constructor(
+    private val insertTransactionModelUseCase: InsertTransactionModelUseCase,
+    private val fetchTransactionModelUseCase: FetchTransactionModelUseCase,
+    private val getTransactionListUseCase: GetTransactionListUseCase
+) : ViewModel() {
+
+    private val LAST_MONTH = 2592000000L
+
+    private val _actions = Channel<Action>(Channel.BUFFERED)
+    val actions = _actions.receiveAsFlow()
+
+    val adapterData: MutableList<ChartItem> = mutableListOf()
+
+    val navigation: NavigationLiveData<ChartNavigationEvent> =
+        NavigationLiveData()
+
+    fun createNotDuplicatedTransactionModel(list: List<TransactionModel>): List<ChartCategoryModel> {
+
+        val categorySum: MutableMap<TransactionCategoryModel, Int> = HashMap()
+        val chartCategoryModelList = mutableListOf<ChartCategoryModel>()
+
+        list.forEach {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
```

Comment:
If you are using any method which required higher API then you have to provide also a way to do same thing for lower API's in another way - so please, handle that in `else` block of that `if`